### PR TITLE
Refactor WAMI Viewer frame source classes

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
Rewrite the internal logic of vpFileDataSource using Qt, rather than a mix of STL, VTK and kwsys, and refactor the API to use Qt conventions. Overhaul vpFrameMap, making use of Qt data types. Refactor users as needed.

This produces a cleaner and more consistent implementation, and will make some future work easier.

### CODE REVIEW

File: Applications/VpView/vpViewCore.cxx [hunk 1]
```diff
@@ -909,29 +909,30 @@ void vpViewCore::exportSceneElementsToFile()
 //-----------------------------------------------------------------------------
 void vpViewCore::exportImageTimeStampsToFile()
 {
-  QString filter = "*.txt;;";
-  vgFileDialog fileDialog(0, tr("Export Image Timestamps"), QString(), filter);
+  vgFileDialog fileDialog{qApp->activeWindow(), "Export Image Timestamps"};
   fileDialog.setObjectName("ExportImageTimeStamps");
   fileDialog.setAcceptMode(QFileDialog::AcceptSave);
+  fileDialog.setNameFilters({"Text files (*.txt)", "All files (*)"});
```
[the remark this answers is not in the document]
Comment (mwoehlke-kitware) [reply Applications/VpView/vpViewCore.cxx]: Sigh... pretty sure that's a VS bug that we've seen before. Try adding another set of `{}`s and let me know if that works. Pretty sure we've had to do that other places...

File: Applications/VpView/vpFrameMap.h [hunk 2]
```diff
@@ -40,12 +40,12 @@ class vpFrame
       }
     }
 
-  void set(unsigned int index, vgTimeStamp time,
-           const vtkMatrix4x4* homography = 0);
+  void set(int index, vgTimeStamp time,
+           const vtkMatrix4x4* homography = nullptr);
 
-  unsigned int Index;
+  int Index = 0;
```
[the remark this answers is not in the document]
Comment (mwoehlke-kitware) [reply Applications/VpView/vpFrameMap.h]: Pretty sure this was for consistency with `vpFileDataSource`, which used/uses `int` for indexing. Basically, changing it resulted in fewer total casts / implicit sign conversions.

File: Applications/VpView/vpViewCore.cxx (hunk 1 above)
[the remark this answers is not in the document]
Comment (RustyBlue) [reply Applications/VpView/vpViewCore.cxx]: `fileDialog.setNameFilters({ {"Text files (*.txt)", "All files (*)"} });`, did not work - is that what you intended for me to try?

error C2664: 'void QFileDialog::setNameFilters(const QStringList &)': cannot convert argument 1 from 'initializer list' to 'const QStringList &'
note: Reason: cannot convert from 'initializer list' to 'const QStringList'
note: No constructor could take the source type, or constructor overload resolution was ambiguous

**Comment by RustyBlue** [comment]
As just discussed - the plan is get hold off on merging this (once approved) until we've moved to Qt5, due to the initializer list issue on Windows.

**Comment by mwoehlke-kitware** [comment]
> the plan is get hold off on merging this (once approved) until we've moved to Qt5

#47 is merged, so... is this approved?

**Pull request merged**